mGui/core/__init__.py

```diff
@@ -122,19 +122,27 @@ def __init__(self, key=None, **kwargs):
         # key is our internal name
         self.key = self.widget.split("|")[-1]
 
+        # Used to track if the control was created in a modal context.
+        # Note that unlike Nested.modal, this is a 'private' field
+        self._is_modal = False
```

Comment:
Doesn't look like we're actually using this anywhere. All the checks seem to be against the public `modal` attribute instead.